### PR TITLE
#54 create UPDATE API User endpoint

### PR DESCRIPTION
**Pull request by pbgnz**
### Issue #54 
[Create UPDATE API User endpoint](https://github.com/jacobrs/SOEN341/issues/54)

### Description
- Created the update User API endpoint
- Made the reducers for the update users
- Fixed some linting issues

### CODE REVIEW

**Comment by pbgnz** [comment]
I can't run the ava tests right now on my PC. I need someone to show me how to properly download the dependencies of the app on windows. The tests for the update endpoint will follow after I properly download all the dependencies.

Also, the update method might need to be tweaked according to the update user info form (which has not been created yet)

File: client/modules/User/UserReducer.js [hunk 1]
```diff
@@ -9,6 +9,10 @@ const UserReducer = (state = initialState, action) => {
       return {
         data: [action.user, ...state.data],
       };
+    case UPDATE_USER :
+      return {
+        data: [action.user, ...state.data],
+      };
```
Comment (simon-bourque) [inline client/modules/User/UserReducer.js]: Make it similar to the LOGIN_USER case like for what to modify in the store

Comment (simon-bourque) [inline client/modules/User/UserReducer.js]:
```
      return {
        user,
      };
```

Comment (simon-bourque) [inline client/modules/User/UserReducer.js]: or user : action.user

Comment (pbgnz) [inline client/modules/User/UserReducer.js]:
```
    case UPDATE_USER : {
      const user = (action.response.statusCode === 200) ? action.response.user : null;

      console.log(user);

      return {
        user,
      };
    }
```

**Comment by jacobrs** [comment]
@pbgnz feel free to merge this to master

**Pull request merged**